modAionImpl/src/org/aion/zero/impl/db/TransactionStore.java

```diff
@@ -100,9 +139,57 @@ public void commit() {
     public void close() throws IOException {
         lock.writeLock().lock();
         try {
-            source.close();
```

Comment:
since they use the same db, I think they can be closed in the same try; the implementation for the store functionality should be verified for exceptions thrown on close when the database is already closed

Reply:
I've tested, and closing both does not throw an exception.